src/Payum/Paypal/ProHosted/Nvp/Action/Api/CreateButtonPaymentAction.php

```diff
@@ -0,0 +1,43 @@
+<?php
+namespace Payum\Paypal\ProHosted\Nvp\Action\Api;
+
+use Payum\Core\Bridge\Spl\ArrayObject;
+use Payum\Core\Exception\LogicException;
+use Payum\Core\Reply\HttpRedirect;
+use Payum\Paypal\ProHosted\Nvp\Request\Api\CreateButtonPayment;
+use Payum\Core\Exception\RequestNotSupportedException;
+
+class CreateButtonPaymentAction extends BaseApiAwareAction
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function execute($request)
+    {
+        /** @var $request CreateButtonPayment */
+        RequestNotSupportedException::assertSupports($this, $request);
+        $model = ArrayObject::ensureArrayObject($request->getModel());
+
+        $model->validateNotEmpty([
+            'subtotal',
+            'currency_code',
+        ]);
+
+        $result = $this->api->doCreateButton((array) $model);
+        $model->replace((array) $result);
+
+        if ($model['EMAILLINK'] != null) {
+            throw new HttpRedirect($model['EMAILLINK']);
+        } else {
+            throw new LogicException('Paypal :'.$model['L_SEVERITYCODE0'].' Code '.$model['L_ERRORCODE0'].': '.$model['L_SHORTMESSAGE0'].'. '.$model['L_LONGMESSAGE0']);
```

Comment:
this is not how payum works, I think you should do nothing in this case and you'll be redirected to done action. there you can get the payment status and it will be failed.